### ISSUE TITLE
Chrome 50 no longer allows geolocation on unsecure pages (http)

### ISSUE THREAD

**Issue by ryanseddon** [Modernizr/Modernizr#1966]
Just saw the [news for chrome 50](https://developers.google.com/web/updates/2016/04/geolocation-on-secure-contexts-only) that geolocation is not going to work on http pages but it still returns true for our test `'geolocation' in navigator` and infact will only show an unhelpful warning in chrome only when you request the position.

![screen shot 2016-04-30 at 4 26 23 pm](https://cloud.githubusercontent.com/assets/143402/14934309/59092fa0-0ef0-11e6-9af9-47e92e5587bc.png)

I've added a [comment to a bug report](https://bugs.chromium.org/p/chromium/issues/detail?id=605938#c3) saying the warning is too ambiguous and they've mentioned that they'll throw an error on `navigator.geolocation.getCurrentPosition()` calls but it'd be ideal if `'geolocation' in navigator` just returned false instead.

Tracking issue here until we know what direction they'll be taking.

**Comment by stucox** [comment]
cc @rbyers, who tends to be helpful with this stuff.

**Comment by RByers** [comment]
Thanks for the report, I agree the message needs to be fixed and I added comments to the bug to that effect.

Trying to find a link now, but I think the general consensus is that feature detection should indicate what APIs a browser supports, not what is currently enabled.  I guess it's fuzzy in this case because it'll never be enabled if the document was loaded via http.  But I could imagine subtler cases that are more dynamic.  For example, maybe the site is loaded over https, but later loads some script with http (and so now contains mixed content and is no longer secure).  We wouldn't want to have the API suddenly disappear.

I think detecting this case is really what the [permissions API](https://developer.mozilla.org/en-US/docs/Web/API/Permissions_API) is for.  Is that good enough for the Modernizr use case?  Or is that API being async a problem?

/cc @mounirlamouri @marcoscaceres

**Comment by mounirlamouri** [comment]
FWIW, the Permissions API already does that on Chrome:
- go to http://example.com
- run `navigator.permissions.query({name:'geolocation'}).then(function(p) { console.log(p.state); })` in the console
- you should see `"denied"`

Hopefully Firefox will do the same when they disable geolocation on non-secure origins.

**Comment by RByers** [comment]
Yep, thanks I saw that (see my [test page](http://output.jsbin.com/kafayu)).  

I think the question here is whether it's desirable for `modernizer.geolocation` to still be `true` in this case, or if Modernizr should be using the permissions API to report that geolocation is in fact unavailable.  I think it's Modernizr's job is to report what APIs are available (a static property), not whether or not the underlying feature is going to be useful or not (a potentially dynamic property), right?  This has come up before in the context of the [Touch Events API](https://hacks.mozilla.org/2013/04/detecting-touch-its-the-why-not-the-how/), which is messier than geolocation due to web compat concerns.

**Comment by ryanseddon** [comment]
I think we could utilise the permissions api as these features migrate to https only. This was we can at least get it right at load time then rely on the browser throwing a useful message if they do something to effect the permissions e.g. load a http resource.
